reana_commons/validation/parameters.py

```diff
@@ -0,0 +1,454 @@
+            steps_used = cmd_param_steps_mapping[parameter]
+            self.workflow_params_warnings.append(
+                {
+                    "type": "warning",
+                    "message": '{type} parameter "{parameter}" found on step{s} "{steps}" is not defined in input parameters.'.format(
+                        type=workflow_type.capitalize(),
+                        parameter=parameter,
+                        steps=", ".join(steps_used),
+                        s="s" if len(steps_used) > 1 else "",
+                    ),
+                }
+            )
+
+    def _validate_misused_parameters_in_steps(
+        self, param_steps_mapping, cmd_param_steps_mapping, workflow_type
+    ):
+        """Validate not used command parameters and not defined input parameters after checking each step of workflow definition.
+
+        :param param_steps_mapping: Mapping between parameters in workflow definition and its step.
+        :param cmd_param_steps_mapping: Mapping between command parameters and its step.
+        :param workflow_type: Workflow type being checked.
+        """
+        command_parameters = set(cmd_param_steps_mapping.keys())
+        workflow_params = set(param_steps_mapping.keys())
+        for parameter in command_parameters:
+            cmd_param_steps = cmd_param_steps_mapping[parameter]
+            param_steps = param_steps_mapping[parameter]
+            steps_diff = cmd_param_steps.difference(param_steps)
+            if steps_diff:
+                self.workflow_params_warnings.append(
+                    {
+                        "type": "warning",
+                        "message": '{type} parameter "{parameter}" found on step{s} "{steps}" is not defined in input parameters.'.format(
+                            type=workflow_type.capitalize(),
+                            parameter=parameter,
+                            steps=", ".join(steps_diff),
+                            s="s" if len(steps_diff) > 1 else "",
+                        ),
+                    }
+                )
+        for parameter in workflow_params:
+            param_steps = param_steps_mapping[parameter]
+            cmd_param_steps = cmd_param_steps_mapping[parameter]
+            steps_diff = param_steps.difference(cmd_param_steps)
+            if steps_diff:
+                self.workflow_params_warnings.append(
+                    {
+                        "type": "warning",
+                        "message": '{type} input parameter "{parameter}" found on step{s} "{steps}" does not seem to be used.'.format(
+                            type=workflow_type.capitalize(),
+                            parameter=parameter,
+                            steps=", ".join(steps_diff),
+                            s="s" if len(steps_diff) > 1 else "",
+                        ),
+                    }
+                )
+
+
+class SerialParameterValidator(ParameterValidatorBase):
```

Comment:
The same naming comment here, `ParameterValidator` prefix would call for `ParameterValidatorSerial`, `ParameterValidatorYadage`, etc.